ext/random/engine_user.c

```diff
@@ -21,6 +21,9 @@
 #include "php.h"
 #include "php_random.h"
 
+#include "ext/spl/spl_exceptions.h"
```

Comment:
bleh are we really going to use SPL exceptions?

Reply:
There's precedent for RuntimeException in the random extension:

https://github.com/php/php-src/blob/dfbe96489d1eccc84470cea27d3ad3cebc3fcd66/ext/random/randomizer.c#L155

But I don't have a strong opinion with regard to the types of exceptions thrown, so I'm happy to adjust this (e.g. to plain Exception/Error).

Reply:
Currently, `random_int()` and `random_bytes()` throw a top-level `\Exception`, which makes it difficult to properly utilize exceptions.

The `ext/spl` exception has already been used by other extensions and is one of the few exceptions that can be used properly, so `ext/random` makes use of it.